examples/cpp/31_oklt_v3_moving_avg/main.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+#include <occa.hpp>
+#include <vector>
+#include "constants.h"
+
+std::vector<float> buildData(std::size_t size,
+                             float initialValue,
+                             float fluctuation)
+{
+    std::vector<float> buffer(size);
+    float currentValue = initialValue;
+    float longIncrement = 1.0f;
+    float fluctuationIncrement = fluctuation;
+    for(std::size_t i = 0; i < buffer.size(); ++i) {
+        buffer[i] = currentValue;
+        fluctuationIncrement = -fluctuationIncrement;
+        if(i % WINDOW_SIZE == 0) {
+            longIncrement = -longIncrement;
+        }
+        currentValue += longIncrement + fluctuationIncrement;
+    }
+    return buffer;
+}
+
+std::vector<float> goldMovingAverage(const std::vector<float> &hostVector) {
+    std::vector<float> result(hostVector.size() - WINDOW_SIZE);
+    for(std::size_t i = 0; i < result.size(); ++i) {
+        float value = 0.0f;
+        for(std::size_t j = 0; j < WINDOW_SIZE; ++j) {
+            value += hostVector[i + j];
+        }
+        result[i] = value / WINDOW_SIZE;
+    }
+    return result;
+}
+
+bool starts_with(const std::string &str, const std::string &substring) {
+    return str.rfind(substring, 0) == 0;
+}
+
+occa::json getDeviceOptions(int argc, const char **argv) {
```

Comment:
Just a minor, there is a type of implementation, occa::json parseArgs(int argc, const char **argv), existing in other examples for the argument parsing. I think it might be better to follow that in case the function is consolidated into a "common" method and being called in the future.

Reply:
Got it. I think I will leave this to other reviewers for comments. Either way, perhaps this could be a future cleanup task to consolidate common methods.